homeassistant/bootstrap.py

```diff
@@ -62,16 +61,13 @@ def setup_component(hass: core.HomeAssistant, domain: str,
 def _handle_requirements(hass: core.HomeAssistant, component,
                          name: str) -> bool:
     """Install the requirements for a component."""
-    if hass.config.skip_pip or not hasattr(component, 'REQUIREMENTS'):
+    if hass.config.skip_pip:
         return True
 
-    for req in component.REQUIREMENTS:
```

Comment:
Why not have a standalone code for looping through the requirements and
logging the error? A handle requirements method just doesn't add any value?

On Mon, Aug 8, 2016, 22:47 Johann Kellerman <notifications@github.com>
wrote:

> In homeassistant/bootstrap.py
> <https://github.com/home-assistant/home-assistant/pull/2765#discussion_r74001138>
> :
>
> >          return True
> >
> > -    for req in component.REQUIREMENTS:
>
> Ok, instead of util.util I will rather move both handle_requirements and
> mount_lib back to bootstrap and just make sure we can call them from
> script
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2765/files/746e11ce1da875820b39e792dc0d11982d94e7ca#r74001138>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABYJ2gm0n9myYlg6R2gjKt4QDnbZyZ6Nks5qeBSHgaJpZM4Jfthk>
> .
>